test/rules.js

```diff
@@ -175,1064 +177,57 @@ describe('Basics', () => {
-            },
-        ],
-        'deliverer-note': [
-            { doc: 'sotd/note-deliverer.html', config: { status: 'WG-NOTE' } },
-            {
-                doc: 'sotd/note-deliverer-bad.html',
-                config: { status: 'WG-NOTE' },
-                errors: ['sotd.deliverer-note.not-found'],
-            },
-        ],
-        'candidate-review-end': [
-            {
-                doc: 'metadata/cr-mediacapture-streams.html',
-                config: { status: 'CR' },
-            },
-            {
-                doc: 'metadata/cr-mediacapture-streams.html',
-                config: { status: 'CR', editorial: true },
-                warnings: ['sotd.candidate-review-end.editorial'],
-            },
-            { doc: 'sotd/cr-end.html', config: { status: 'CR' } },
-            {
-                doc: 'sotd/cr-end-27days.html',
-                config: { status: 'CR' },
-                errors: ['sotd.candidate-review-end.not-found'],
-            },
-            {
-                doc: 'sotd/cr-end-multiple.html',
-                config: { status: 'CR' },
-                warnings: ['sotd.candidate-review-end.multiple-found'],
-            },
-            {
-                doc: 'sotd/cr-end-nodate.html',
-                config: { status: 'CR' },
-                errors: ['sotd.candidate-review-end.found-not-valid'],
-            },
-        ],
-    },
-    heuristic: {
-        'date-format': [
-            { doc: 'heuristic/dates.html' },
-            {
-                doc: 'heuristic/bad-dates.html',
-                errors: [
-                    'heuristic.date-format.wrong',
-                    'heuristic.date-format.wrong',
-                ],
-            },
-            { doc: 'heuristic/dated-url.html' },
-        ],
-    },
-    validation,
-};
-
 // start an server to host doc, response to sr.url requests
 const app = express();
 app.use('/docs', express.static(pth.join(__dirname, 'docs')));
-const expressServer = app.listen(PORT, () => {});
+
+// use express-handlebars
+app.engine(
+    'handlebars',
+    exphbs.engine({
+        defaultLayout: pth.join(__dirname, './doc-views/layout/TR'),
+        layoutsDir: pth.join(__dirname, './doc-views'),
+        partialsDir: pth.join(__dirname, './doc-views/partials/'),
+    })
+);
+app.set('view engine', 'handlebars');
+app.set('views', pth.join(__dirname, './doc-views'));
+
+app.get('/doc-views/:docType/:track/:profile', (req, res) => {
+    const { rule, type } = req.query;
+    // get data for template from json (.js)
+    const data = require(pth.join(
+        __dirname,
+        `./doc-views/${req.params.docType}/${req.params.track}/${req.params.profile}.js`
+    ));
+
+    let finalData;
+    if (type.startsWith('good')) {
+        finalData = data[type];
+    } else {
+        if (!rule)
+            res.send(
+                '<h1>Error: please add the parameter "rule" in the URL </h1>'
+            );
+        if (!type)
```

Comment:
With the `startsWith` above, that `if` is useless as there will be an exception before. This might not be a problem if we change that code.